core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosClusterScheduler.scala

```diff
@@ -422,6 +422,37 @@ private[spark] class MesosClusterScheduler(
     desc.schedulerProperties.get("spark.cores.max").map { v =>
       options ++= Seq("--total-executor-cores", v)
     }
+
+    /**
+      * The properties to pass along from Spark Submit to the Mesos Scheduler.
+      * http://spark.apache.org/docs/1.5.1/configuration.html#available-properties
+      */
+    val propertiesToPropagate = Set(
+      "spark.driver.extraClassPath",
```

Comment:
My app jar was duplicated in the classpath due to the `spark.jars` property, then I wondered about which settings you care about when "_customizing_" a job vs infrastructure implied settings from the config files. Stripping `spark.jars` would probably be fine, but I'm not familiar enough with the framework to know of any other potential conflicts.